libcontainer/container_linux.go

```diff
@@ -1863,7 +1863,7 @@ func (c *linuxContainer) currentOCIState() (*specs.State, error) {
 	if err != nil {
 		return nil, err
 	}
-	state.Status = status.String()
+	state.Status = specs.ContainerState(status.String())
```

Comment:
Notice that `status.String()` can return states that are not in the OCI runtime spec (`"pausing"`, `"paused"`, `"unknown"`), and vice versa: the runtime spec defines `"creating"` which does not map to a libcontainer state ; https://github.com/opencontainers/runc/blob/ceefc3fe4ecb31fa89ed50ecb2bc0efc7dfac683/libcontainer/container.go#L31-L46


How should we deal with those?

Reply:
I think it's fine for runc to have a superset of statuses as defined by OCI spec. It's also fine to never return some statuses, like "creating".